### PR TITLE
[ARCTIC-667][AMS]: terminal support iceberg connector

### LINKED ISSUE #667: [Feature]: AMS-Terminal Should support iceberg connector
### Description

when ams support iceberg type table. terminal should support iceberg 

### Use case/motivation

_No response_

### Describe the solution

_No response_

### Subtasks

_No response_

### Related issues

#262 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #724: [Bug]: terminal doesn't execute sql really when sql is not end with ;
### What happened?

image

### Affects Versions

0.3.2

### What engines are you seeing the problem on?

AMS, Spark

### How to reproduce

execute some sql not end with semicolon 

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #738: [Improvement]:   quickstart in local mode, the log doesn't have date
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

image

### How should we improve?

add date  before every log string

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

fix #667  fix #724  fix #738

## Brief change log

- add support for iceberg catalog in terminal
- auto add `;` when terminal execute script without `;` end ;

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes )
  - If yes, how is the feature documented? ( docs )

### CODE REVIEW

**Comment by zhoujinsong** [comment]
@baiyangtx Could you rename the PR's title to match pattern `[ARCTIC-{issue_number}][{module}]{pr_description}`?

**Comment by baiyangtx** [comment]
> @baiyangtx Could you rename the PR's title to match pattern `[ARCTIC-{issue_number}][{module}]{pr_description}`?

done

File: core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java [hunk 1]
```diff
@@ -57,36 +57,39 @@ public String name() {
...
   public List<String> listDatabases() {
     if (icebergCatalog instanceof SupportsNamespaces) {
-      return tableMetaStore.doAs(() -> ((SupportsNamespaces)icebergCatalog).listNamespaces(Namespace.empty()).stream()
+      return tableMetaStore.doAs(() -> ((SupportsNamespaces) icebergCatalog).listNamespaces(Namespace.empty()).stream()
           .map(namespace -> namespace.level(0)).distinct().collect(Collectors.toList()));
     } else {
-      throw new UnsupportedOperationException(String.format("Iceberg catalog: %s do now implement SupportsNamespaces",
-       icebergCatalog.getClass().getName()));
+      throw new UnsupportedOperationException(String.format(
+          "Iceberg catalog: %s do now implement SupportsNamespaces",
```
Comment (Aireed) [inline core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java:78]: `"Iceberg catalog: %s do now implement SupportsNamespaces",`
do now  --> does not

File: core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java [hunk 2]
```diff
@@ -57,36 +57,39 @@ public String name() {
...
     if (icebergCatalog instanceof SupportsNamespaces) {
       tableMetaStore.doAs(() -> {
-        ((SupportsNamespaces)icebergCatalog).createNamespace(Namespace.of(databaseName));
+        ((SupportsNamespaces) icebergCatalog).createNamespace(Namespace.of(databaseName));
         return null;
       });
     } else {
-      throw new UnsupportedOperationException(String.format("Iceberg catalog: %s do now implement SupportsNamespaces",
+      throw new UnsupportedOperationException(String.format(
+          "Iceberg catalog: %s do now implement SupportsNamespaces",
```
Comment (Aireed) [inline core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java:92]: `"Iceberg catalog: %s do now implement SupportsNamespaces",`
same as last one

File: core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java [hunk 3]
```diff
@@ -95,21 +98,22 @@ public void createDatabase(String databaseName) {
...
     if (icebergCatalog instanceof SupportsNamespaces) {
       tableMetaStore.doAs(() -> {
-        ((SupportsNamespaces)icebergCatalog).dropNamespace(Namespace.of(databaseName));
+        ((SupportsNamespaces) icebergCatalog).dropNamespace(Namespace.of(databaseName));
         return null;
       });
     } else {
-      throw new UnsupportedOperationException(String.format("Iceberg catalog: %s do now implement SupportsNamespaces",
+      throw new UnsupportedOperationException(String.format(
+          "Iceberg catalog: %s do now implement SupportsNamespaces",
```
Comment (Aireed) [inline core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java:106]: "Iceberg catalog: %s do now implement SupportsNamespaces",
`do now`  same as last one

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalManager.java [hunk 4]
```diff
@@ -91,34 +93,35 @@ public String executeScript(String terminalId, String catalog, String script) {
...
     String sessionId = getSessionId(terminalId, metaStore);
+    String catalogType = CatalogUtil.isIcebergCatalog(catalog) ? "iceberg" : "arctic";
+    Configuration configuration = new Configuration();
+    configuration.setInteger(SessionConfigOptions.FETCH_SIZE, resultLimits);
+    configuration.set(SessionConfigOptions.CATALOGS, Lists.newArrayList(catalog));
+    configuration.set(SessionConfigOptions.catalogConnector(catalog), catalogType);
+    configuration.set(SessionConfigOptions.CATALOG_URL_BASE, String.format(
+        "thrift://%s:%d/",
+        ArcticMetaStore.conf.getString(ArcticMetaStoreConf.THRIFT_BIND_HOST),
+        ArcticMetaStore.conf.getInteger(ArcticMetaStoreConf.THRIFT_BIND_PORT)));
```
Comment (Aireed) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalManager.java:104]: ` configuration.set(SessionConfigOptions.CATALOG_URL_BASE, String.format(
        "thrift://%s:%d/",
        ArcticMetaStore.conf.getString(ArcticMetaStoreConf.THRIFT_BIND_HOST),
        ArcticMetaStore.conf.getInteger(ArcticMetaStoreConf.THRIFT_BIND_PORT)));`
shall we consider AMS ha mode??

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalSessionContext.java [hunk 5]
```diff
@@ -188,6 +192,10 @@ public ExecutionStatus get() {
           TerminalSession session = lazyLoadSession(this);
           executionResult.appendLog("fetch terminal session: " + sessionId);
           executionResult.appendLogs(session.logs());
+          for (String key : session.configs().keySet()) {
+            executionResult.appendLog("session configuration: " + key + " => " + session.configs().get(key));
```
Comment (Aireed) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalSessionContext.java:196]: should we add date before log?

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/kyuubi/KyuubiTerminalSessionFactory.java [hunk 6]
```diff
@@ -103,25 +108,24 @@ public TerminalSession create(TableMetaStore metaStore, Configuration configurat
...
-      String catalogUrl = configuration.get(SessionConfigOptions.catalogUrl(catalog));
-      String type = configuration.get(SessionConfigOptions.catalogType(catalog));
+    Map<String, String> sparkConf = SparkContextUtil.getSparkConf(configuration);
+    sparkConf.forEach((k, v) -> connectionParams.getHiveVars().put(k, v));
 
-      connectionParams.getHiveVars().put("spark.sql.catalog." + catalog, ArcticSparkCatalog.class.getName());
-      connectionParams.getHiveVars().put("spark.sql.catalog." + catalog + ".url", catalogUrl);
-    }
     String kyuubiJdbcUrl = getConnectionUrl(connectionParams);
     logMessage(logs, "try to create a kyuubi connection via url: " + kyuubiJdbcUrl);
```
Comment (Aireed) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/kyuubi/KyuubiTerminalSessionFactory.java:115]: lack of date before log string

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Comment by zhoujinsong** [comment]
Do you resolve #738  in this PR too?
If yes, you may want to link it in your description. @baiyangtx
